### PR TITLE
Unify bazel versions

### PR DESCRIPTION
**Pull request by jtattermusch**
and start using tools/bazel.sh wrapper where possible.

### CODE REVIEW

File: tools/remote_build/README.md [hunk 1]
```diff
@@ -17,22 +17,27 @@ and tests run by Kokoro CI.
 
 ## Running remote build manually from dev workstation
 
+*At the time being, tools/bazel.sh is used instead of invoking "bazel" directly
```
Comment (gnossen) [inline tools/remote_build/README.md]: Would it be a good idea to hoist this somewhere higher? Perhaps the top-level README?

Comment (jtattermusch) [reply tools/remote_build/README.md]: This is mostly meant as a heads up for the team (others can't run the manual remote bazel runs anyway) so we'd announce internally.

**Comment by jtattermusch** [comment]
Known failures:
https://github.com/grpc/grpc/issues/18967
https://github.com/grpc/grpc/issues/18416

**Comment by jtattermusch** [comment]
Looks like there's still some problem with the ubsan RBE run.
https://source.cloud.google.com/results/invocations/bd5e84a7-f517-490e-923e-9ed3c703630d/log

**Comment by AustinSchuh** [comment]
Little know fact, on Linux and OSX, ```/usr/bin/bazel``` is a shell script.  It first looks in ```//tools/bazel``` (note, no .sh at the end).  If it finds it, it calls it and ignores the host version.

So, if you call your ```//tools/bazel.sh``` script ```tools/bazel```, any call to bazel will call your sandboxed bazel version instead.

**Comment by jtattermusch** [comment]
> Little know fact, on Linux and OSX, `/usr/bin/bazel` is a shell script. It first looks in `//tools/bazel` (note, no .sh at the end). If it finds it, it calls it and ignores the host version.
> 
> So, if you call your `//tools/bazel.sh` script `tools/bazel`, any call to bazel will call your sandboxed bazel version instead.

Thanks for the recommendation, we will try that! It that feature officially documented and is there a way to make it work on windows too?

**Comment by AustinSchuh** [comment]
@jtattermusch I'm unsure on Windows, though I'd recommend using the same path regardless.  One of my coworkers added the OSX and Linux support.  If I had access to a windows box, I'd go try it for you.

The best docs I can find are in the source.  https://github.com/bazelbuild/bazel/blob/master/scripts/packages/bazel.sh

**Comment by jtattermusch** [comment]
Known failures:
https://github.com/grpc/grpc/issues/18967
https://github.com/grpc/grpc/issues/18608
https://github.com/grpc/grpc/issues/18416

**Comment by jtattermusch** [comment]
> @jtattermusch I'm unsure on Windows, though I'd recommend using the same path regardless. One of my coworkers added the OSX and Linux support. If I had access to a windows box, I'd go try it for you.
> 
> The best docs I can find are in the source. https://github.com/bazelbuild/bazel/blob/master/scripts/packages/bazel.sh

We'll look into utilizing https://github.com/bazelbuild/bazel/blob/master/scripts/packages/bazel.sh#L77 in a followup PR.

**Pull request merged**